packages/components/app/styles/components/form/text-input.scss

```diff
@@ -0,0 +1,127 @@
+//
+
+// "BASE" CONTROL
+
+.hds-form-text-input {
+  border: 1px solid var(--token-color-palette-neutral-400);
+  border-radius: 5px;
+  box-shadow: var(--hds-elevation-inset-box-shadow);
+  color: var(--token-color-foreground-primary);
+  font-family: var(--token-typography-body-200-font-family);
+  font-size: var(--token-typography-body-200-font-size);
+  line-height: var(--token-typography-body-200-line-height);
+  padding: 7px;
+  width: 100%;
+  max-width: 100%;
+
+  // PLACEHOLDER
+
+  ::placeholder {
+    color: var(--token-color-foreground-faint);
+  }
+
+  // STATUS
+
+  &:hover,
+  &.mock-hover {
+    border-color: var(--token-color-palette-neutral-500);
+  }
+
+  // focus (same for all the states)
+  // TODO add handling of focus-visible
+  &:focus,
+  &.mock-focus {
+    border-color: var(--token-color-focus-action-internal);
+    // TODO: Safari doesn't apply a rounded border
+    outline: 2px solid var(--token-color-focus-action-external);
+    outline-offset: 0px;
+  }
+
+  &:active,
+  &.mock-active {
+    border-color: var(--token-color-focus-action-internal);
+  }
+
+  // DISABLED
+
+  &:disabled {
+    background-color: var(--token-color-surface-interactive-disabled);
+    border-color: var(--token-color-border-primary);
+  }
+
+  // INVALID
+
+  &.hds-form-text-input--is-invalid {
+    border-color: var(--token-color-foreground-critical);
+  }
+}
+
+
+
+// "TYPE" CUSTOMIZATION
+
+.hds-form-text-input {
+
+  // DATE/TIME
+
+  &[type="date"],
+  &[type="time"] {
+
+    // browsers set a specific width for these controls, we want to keep it
+    width: initial;
+
+    // show the native icon dimmed if disabled (hidden in Chrome)
+    &[disabled]::-webkit-calendar-picker-indicator,
+    &:disabled::-webkit-calendar-picker-indicator {
+        visibility: visible;
+        opacity: 0.5;
+    }
+
+    // show the icon if readonly
+    &[readonly]::-webkit-calendar-picker-indicator {
+        visibility: visible;
+    }
+  }
+
+  // we override the default icon with the Flight corresponding one
+  // notice: the original in Chrome has two assets, one for light and one for dark mode, and uses a special syntax, but apparently it doesn't work if used in a stylesheet
+  &[type="date"] {
+    &::-webkit-calendar-picker-indicator {
+      // notice: the icon color is hardcoded here!
+      background-image: url("data:image/svg+xml,%3csvg viewBox='0 0 16 16' xmlns='http://www.w3.org/2000/svg'%3e%3cpath d='M11.5.75a.75.75 0 00-1.5 0V1H6V.75a.75.75 0 00-1.5 0V1H3.25A2.25 2.25 0 001 3.25v9.5A2.25 2.25 0 003.25 15h9.5A2.25 2.25 0 0015 12.75v-9.5A2.25 2.25 0 0012.75 1H11.5V.75zm-7 2.5V2.5H3.25a.75.75 0 00-.75.75V5h11V3.25a.75.75 0 00-.75-.75H11.5v.75a.75.75 0 01-1.5 0V2.5H6v.75a.75.75 0 01-1.5 0zm9 3.25h-11v6.25c0 .414.336.75.75.75h9.5a.75.75 0 00.75-.75V6.5z' fill-rule='evenodd' clip-rule='evenodd' fill='%233B3D45'/%3e%3c/svg%3e");
+      background-position: center center;
+      background-size: 16px;
```

Comment:
Thought: I've just realised we have this 'icon size' value in so many places we might want to convert it into a variable, in this file at least.

Reply:
Yes, the idea is to have a set of component-level design tokens, specific for the `form controls`, once the design specs are finalized, so we can share them with DevDot too.